### DISCUSSION TITLE
How to exclude all java-Files that contain a @Generated-Annotation at class-level?

### DISCUSSION THREAD

**Discussion by AlBundy33** [checkstyle/checkstyle#14773] [Q&A] [answered]
moved from #14770 

We have one common checkstyle-config for all projects and therefore we can not exclude specific paths.
But all of the generated java-Files have a @Generated-Annotation at least at class level
```
// some comment
import javax.annotation.Generated;
@Generated
public class Foo {
   public void bar() {
   }
}
```
This genereted files have checkstyle errors everywhere. 😄 
First error on first line because we expect multiline-comments `/** */` instead of `//`
sometime also forbidden imports.
everytime curlybraces on the wrong line
and so on

therefore I want to exclude those files completly.

I have no idea how to achive this with [SuppressionXpathSingleFilter](https://checkstyle.sourceforge.io/filters/suppressionxpathsinglefilter.html)

I've tested [SuppressWithNearbyTextFilter](https://checkstyle.sourceforge.io/filters/suppresswithnearbytextfilter.html) with this settings but this looks kind of strange.

```
    <module name="SuppressWithNearbyTextFilter">
        <property name="checkPattern" value=".*"/>
        <property name="nearbyTextPattern" value="^@Generated\s*\("/>
        <property name="lineRange" value="-99999"/>
    </module>
    <module name="SuppressWithNearbyTextFilter">
        <property name="checkPattern" value=".*"/>
        <property name="nearbyTextPattern" value="^@Generated\s*\("/>
        <property name="lineRange" value="99999"/>
    </module>
```

Has anyone a better solution?

**Comment by rnveach** [comment]
Please provide an example of a check and working violation you wish to suppress. If you have any not inside TreeWalker, then I suggest you give us that as an example.

**Comment by AlBundy33** [comment]
No problem: [example.zip](https://github.com/checkstyle/checkstyle/files/14931267/example.zip)
If you run the example with
`java -Duser.language=en -jar checkstyle-10.15.0-all.jar -c cs.xml Foo.java`
you'll get this output
```
Starting audit...
[ERROR] C:\cs\Foo.java:1: Line does not match expected header line of '^/\*+$'. [RegexpHeader]
[WARN] C:\cs\Foo.java:1: Missing package-info.java file. [JavadocPackage]
[ERROR] C:\cs\Foo.java:14:18: '{' at column 18 should be on a new line. [LeftCurly]
[ERROR] C:\cs\Foo.java:15:1: Line contains a tab character. [FileTabCharacter]
[ERROR] C:\cs\Foo.java:15:9: Missing a Javadoc comment. [MissingJavadocMethod]
[ERROR] C:\cs\Foo.java:15:27: '{' at column 20 should be on a new line. [LeftCurly]
[ERROR] C:\cs\Foo.java:16:1: Line contains a tab character. [FileTabCharacter]
[ERROR] C:\cs\Foo.java:18:1: Line contains a tab character. [FileTabCharacter]
[ERROR] C:\cs\Foo.java:18:9: Missing a Javadoc comment. [MissingJavadocMethod]
[ERROR] C:\cs\Foo.java:18:27: '{' at column 20 should be on a new line. [LeftCurly]
[ERROR] C:\cs\Foo.java:19:1: Line contains a tab character. [FileTabCharacter]
[ERROR] C:\cs\Foo.java:20:1: Line contains a tab character. [FileTabCharacter]
[ERROR] C:\cs\Foo.java:21:1: Line contains a tab character. [FileTabCharacter]
[ERROR] C:\cs\Foo.java:22:1: Line contains a tab character. [FileTabCharacter]
[ERROR] C:\cs\Foo.java:23:1: Line contains a tab character. [FileTabCharacter]
[ERROR] C:\cs\Foo.java:24:1: Line contains a tab character. [FileTabCharacter]
[ERROR] C:\cs\Foo.java:25: Line is longer than 128 characters (found 140). [LineLength]
[ERROR] C:\cs\Foo.java:25:1: Line contains a tab character. [FileTabCharacter]
[ERROR] C:\cs\Foo.java:26:1: Line contains a tab character. [FileTabCharacter]
Audit done.
Checkstyle ends with 18 errors.
```

All errors are correct but since the Java-File is generated (annotated with @Generated at class-level) I want to ignore the complete file.

**Reply by romani** [reply, marked as answer]
I think that used `SuppressWithNearbyTextFilter` is best form of suppression, while excluding folders and files is not an option.

**Reply by AlBundy33** [reply]
Ok - thanks for your answer. 👍

**Reply by rnveach** [reply]
If you are going to suppress all of checkstyle for these files, then it is probably better to put them into some commonly named folder across all your projects. For example, maven puts all generated source files into `generated-sources`. Otherwise, you have to modify the file to add the comment which in a way defeats the purpose of trying to ignore a generated file.

**Reply by AlBundy33** [reply]
most of the generated classes have the @Generated-annotation but have no common directory to exclude them.
therefore I've searched for a filter that can exclude a class based on the existence of this annotation.

the nearby-filter works but felt kind of strange due to large range that have to be configured to match all files.
But I'm fine with it if this is the way to go.
